kernel/src/sys/sched_mlfq.c

```diff
@@ -0,0 +1,328 @@
+#include <kernel/sched_robin.h>
+#include <kernel/sys.h>
+#include <kernel/timer.h>
+#include <stdlib.h>
+
+uint32_t flag;
+/* Wraps a `process_t*` for round robin purposes.
+ */
+typedef struct _proc_node_t {
+    process_t* process;
+    struct _proc_node_t* next;
+    uint32_t count;
+} proc_node_t;
+
+/* The round robin scheduler is simple and requires only a single circular list
+ * containing candidate processes. By having a `sched_t` as the first member of
+ * the struct, we allow casting `sched_robin_t*`s to `sched_t*`.
+ */
+typedef struct {
+    sched_t sched;
+    proc_node_t* processes1;
+    proc_node_t* processes2;
+    proc_node_t* processes3;
```

Comment:
Do circular lists really make sense for this type of scheduler? Could regular lists be used?